src/v/crypto/random.cc

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "crypto/crypto.h"
+
+#include <openssl/rand.h>
+
+namespace crypto {
+bytes_span<> generate_random(bytes_span<> buf, is_private_rng_t is_private) {
+    if (is_private) {
+        RAND_priv_bytes_ex(nullptr, buf.data(), buf.size(), 0);
+    } else {
+        RAND_bytes_ex(nullptr, buf.data(), buf.size(), 0);
+    }
```

Comment:
OpenSSL docs suggest that these can fail, putting the CSPRNG in an error state. Can/should we check for that somehow?

Reply:
oh good catch